src/typeahead/typeahead-window.ts

```diff
@@ -20,16 +20,18 @@ export interface ResultTemplateContext {
 @Component({
   selector: 'ngb-typeahead-window',
   exportAs: 'ngbTypeaheadWindow',
-  host: {'class': 'dropdown-menu', 'style': 'display: block'},
+  host: {'class': 'dropdown-menu', 'style': 'display: block', 'role': 'listbox'},
   template: `
     <template #rt let-result="result" let-term="term" let-formatter="formatter">
       <ngb-highlight [result]="formatter(result)" [term]="term"></ngb-highlight>
     </template>
     <template ngFor [ngForOf]="results" let-result let-idx="index">
-      <button type="button" class="dropdown-item" [class.active]="idx === activeIdx" 
-        (mouseenter)="markActive(idx)" 
+      <button type="button" class="dropdown-item" role="option"
+        [attr.id]="id + '-' + idx"
```

Comment:
I think that you can use `[id]="..."` instead of `[attr.id]` as buttons should have the `id` property.